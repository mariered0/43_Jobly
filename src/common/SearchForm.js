import React, { useState } from "react";
import { Form, Input, Button, Row, Col } from "reactstrap";
import CompanyList from "../company/CompanyList"

/** Search form 
 * 
 * Used on Company and Job to allow search for companies and jobs.
*/


const SearchForm = ({ search }) => {
    const INITIAL_STATE = {
        term: ""
    }
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const { name, value} = e.target;

        setFormData((formData) => ({
            ...formData,
            [name]: value
        }));
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        search(formData.term);    
    }


  return (
    <div className="container">
      <Form>
        <Row className="row-cols-lg-auto g-3">
          <Col
               md="6" 
               lg="8">
              <Input
                bsSize="lg"
                className="mb-3"
                id="term"
                name="term"
                placeholder="Enter search term..."
                type="text"
                onChange={handleChange}
              />
          </Col>
          <Col>
            <Button size="lg"
                    type="submit"
                    onClick={handleSubmit}>Submit</Button>
          </Col>
        </Row>
      </Form>
    </div>
  );
};

export default SearchForm;
