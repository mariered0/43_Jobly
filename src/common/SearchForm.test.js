import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "./SearchForm";

// it("renders without crashing", () => {
//     render(<SearchForm />);
// });

// it("matches snapshot", () => {
// 	const { asFragment } = render(<SearchForm />);
// 	expect(asFragment()).toMatchSnapshot();
// });