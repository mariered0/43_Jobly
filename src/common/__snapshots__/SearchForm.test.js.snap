// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <form
      class=""
    >
      <div
        class="row-cols-lg-auto g-3 row"
      >
        <div
          class="col-md-6 col-lg-8"
        >
          <input
            class="mb-3 form-control-lg form-control"
            id="term"
            name="term"
            placeholder="Enter search term..."
            type="text"
          />
        </div>
        <div
          class="col"
        >
          <button
            class="btn btn-secondary btn-lg"
            type="submit"
          >
            Submit
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
